### DISCUSSION TITLE
Has user provided issue compiling

### DISCUSSION THREAD

**Discussion by maxrichardson250902** [RosettaCommons/rosetta#166] [Build & Install]
When i try to compile, there seems to be an issue around various option keys leading to this. 

In file included from src/utility/options/keys/AnyOptionKey.hh:23,
                 from src/basic/options/keys/OptionKeys.hh:48,
                 from src/basic/options/keys/packing.OptionKeys.gen.hh:19,
                 from src/protocols/matdes/MatDesGreedyOptMutationMover.cc:56:
src/utility/options/keys/ScalarOptionKey.hh:53:9: note: because 'utility::options::ScalarOptionKey' has user-provided 'utility::options::ScalarOptionKey::ScalarOptionKey(const utility::options::ScalarOptionKey&)'
   53 |         ScalarOptionKey( ScalarOptionKey const & key ) :
      |         ^~~~~~~~~~~~~~~
In file included from src/utility/options/BooleanOption.hh:23,
                 from src/utility/options/OptionCollection.hh:24,
                 from src/basic/options/option.hh:17,
                 from src/protocols/matdes/MatDesGreedyOptMutationMover.cc:58:
src/utility/options/ScalarOption_T_.hh:143:30: note: synthesized method 'utility::options::IntegerOptionKey& utility::options::IntegerOptionKey::operator=(const utility::options::IntegerOptionKey&)' first required here
  143 |                         key_ = option.key_;
      |                         ~~~~~^~~~~~~~~~~~~
In file included from src/basic/options/keys/OptionKeys.hh:59,
                 from src/basic/options/keys/packing.OptionKeys.gen.hh:19,
                 from src/protocols/matdes/MatDesGreedyOptMutationMover.cc:56:
src/utility/options/keys/RealOptionKey.hh: In instantiation of 'utility::options::ScalarOption_T_<K, T>& utility::options::ScalarOption_T_<K, T>::operator=(const utility::options::ScalarOption_T_<K, T>&) [with K = utility::options::RealOptionKey; T = double]':
src/utility/options/RealOption.hh:40:7:   required from here
src/utility/options/keys/RealOptionKey.hh:35:7: warning: implicitly-declared 'utility::options::ScalarOptionKey& utility::options::ScalarOptionKey::operator=(const utility::options::ScalarOptionKey&)' is deprecated [-Wdeprecated-copy]
   35 | class RealOptionKey :
      |       ^~~~~~~~~~~~~
In file included from src/utility/options/keys/AnyOptionKey.hh:23,
                 from src/basic/options/keys/OptionKeys.hh:48,
                 from src/basic/options/keys/packing.OptionKeys.gen.hh:19,
                 from src/protocols/matdes/MatDesGreedyOptMutationMover.cc:56:
src/utility/options/keys/ScalarOptionKey.hh:53:9: note: because 'utility::options::ScalarOptionKey' has user-provided 'utility::options::ScalarOptionKey::ScalarOptionKey(const utility::options::ScalarOptionKey&)'
   53 |         ScalarOptionKey( ScalarOptionKey const & key ) :
      |         ^~~~~~~~~~~~~~~
In file included from src/utility/options/BooleanOption.hh:23,
                 from src/utility/options/OptionCollection.hh:24,
                 from src/basic/options/option.hh:17,
                 from src/protocols/matdes/MatDesGreedyOptMutationMover.cc:58:
src/utility/options/ScalarOption_T_.hh:143:30: note: synthesized method 'utility::options::RealOptionKey& utility::options::RealOptionKey::operator=(const utility::options::RealOptionKey&)' first required here
  143 |                         key_ = option.key_;
      |                         ~~~~~^~~~~~~~~~~~~
In file included from src/basic/options/keys/OptionKeys.hh:63,
                 from src/basic/options/keys/packing.OptionKeys.gen.hh:19,
                 from src/protocols/matdes/MatDesGreedyOptMutationMover.cc:56:
src/utility/options/keys/StringOptionKey.hh: In instantiation of 'utility::options::ScalarOption_T_<K, T>& utility::options::ScalarOption_T_<K, T>::operator=(const utility::options::ScalarOption_T_<K, T>&) [with K = utility::options::StringOptionKey; T = std::__cxx11::basic_string<char>]':
src/utility/options/StringOption.hh:35:7:   required from here
src/utility/options/keys/StringOptionKey.hh:35:7: warning: implicitly-declared 'utility::options::ScalarOptionKey& utility::options::ScalarOptionKey::operator=(const utility::options::ScalarOptionKey&)' is deprecated [-Wdeprecated-copy]
   35 | class StringOptionKey :
      |       ^~~~~~~~~~~~~~~
In file included from src/utility/options/keys/AnyOptionKey.hh:23,
                 from src/basic/options/keys/OptionKeys.hh:48,
                 from src/basic/options/keys/packing.OptionKeys.gen.hh:19,
                 from src/protocols/matdes/MatDesGreedyOptMutationMover.cc:56:
src/utility/options/keys/ScalarOptionKey.hh:53:9: note: because 'utility::options::ScalarOptionKey' has user-provided 'utility::options::ScalarOptionKey::ScalarOptionKey(const utility::options::ScalarOptionKey&)'
   53 |         ScalarOptionKey( ScalarOptionKey const & key ) :
      |         ^~~~~~~~~~~~~~~
In file included from src/utility/options/BooleanOption.hh:23,
                 from src/utility/options/OptionCollection.hh:24,
                 from src/basic/options/option.hh:17,
                 from src/protocols/matdes/MatDesGreedyOptMutationMover.cc:58:
src/utility/options/ScalarOption_T_.hh:143:30: note: synthesized method 'utility::options::StringOptionKey& utility::options::StringOptionKey::operator=(const utility::options::StringOptionKey&)' first required here
  143 |                         key_ = option.key_;
      |                         ~~~~~^~~~~~~~~~~~~

This is just a snapshot of it but it continues until breaking due to many errors.

**Comment by roccomoretti** [comment]
This looks like it's due to increased warning levels (specifically `-Wdeprecated-copy`) on newer compilers. We compile Rosetta with warnings in the development versions to make sure we're not messing things up, but a consequence is that when new compilers come out there's sometimes new warning issues that weren't caught with earlier compilers. -- We try to fix them as they come up, but it sometimes takes a while before we can implement and test a fix. (By the way, which compiler and version are you seeing the issues with?)

For released versions, we typically turn off the warnings, so you probably won't see this if you compile the downloaded version from the RosettaCommons website. The way this is accomplished is by deleting the lines labeled "REMOVE FOR RELEASE" in the `source/tools/build/basic.settings` file. If you're using the Github version, that might be a workaround -- edit your working copy to remove those lines from that file.

**Reply by maxrichardson250902** [reply]
I dont have a REMOVE FOR RELEASE line. 

Trying with the Github version now and it gives this. 
scons: Reading SConscript files ...
Running versioning script ...
Encounter error while executing: git rev-parse "HEAD:source/src/python/PyRosetta/binder"

Encounter error while executing: git rev-parse "HEAD:source/src/python/PyRosetta/binder"
fatal: not a git repository (or any of the parent directories): .git

**Reply by maxrichardson250902** [reply]
thanks for your reply !

**Reply by roccomoretti** [reply]
How are you obtaining Rosetta? If you want just a source bundle, go to https://rosettacommons.org/download/ and download from there. If you want a development version, use `git clone`. I would not recommend downloading a single version snapshot from Github (it might work, but we're not set up to support that).

If you obtain Rosetta from Github, it should have the REMOVE FOR RELEASE lines. If you download it from the rosettacommons.org website, we'll have removed those lines for you already.

If you're still having issues compiling, please clarify how exactly you're obtaining the source code (the details of the steps/commands you used to get it), as well as providing details about which compiler version you're using.

I should also note that the  https://rosettacommons.org/download/  has pre-compiled versions that should work on many systems. It might be worth just using that and skipping the compilation step if you're encountering issues.

**Reply by maxrichardson250902** [reply]
switched to the GitHub version and it is now working, i cant remember where i got the other version from. I will remove the lines in the github version thanks. 

I am hoping to use RifDOCK and the README for that details how a specific branch needs to be used but i cant find the branch on here for Bcov/stable

**Reply by roccomoretti** [reply]
We switched repositories at the start of 2024, and the branches and SHA1 hashes which were in the old repository don't necessarily transfer over.

Digging into the old (private) repository, it looks like the branch `bcov/stable1` only has minor changes from the commit which is SHA1 `2c6d0efdbf7374e2b5627607391c308e46dfddf4` in the new repository. If you clone the RosettaCommons/rosetta repository and do a `git checkout 2c6d0efdbf7374e2b5627607391c308e46dfddf4`, that should get you a code state which is more-or-less identical to the branch `bcov/stable1` in the old repo. 

Actually, for convenience sake I've gone ahead and added an equivalent branch to the current repo.

**Reply by maxrichardson250902** [reply]
Thank you for taking the time to do that and for answering my questions.
